main.sh

```diff
@@ -16,4 +16,106 @@
 #
 # Copyright 2023 The Kepler Contributors
 #
-./common.sh $1
+
+set -eu -o pipefail
+
+PROJECT_ROOT="$(git rev-parse --show-toplevel)"
+declare -r PROJECT_ROOT
+declare -r BIN_DIR="$PROJECT_ROOT/tmp/bin"
+
+source "$PROJECT_ROOT/.env"
+source "$PROJECT_ROOT/lib/utils.sh"
+
+declare -r CTR_CMD=${CTR_CMD:-docker}
+declare -r CLUSTER_PROVIDER=${CLUSTER_PROVIDER:-kind}
+declare -r REGISTRY_PORT=${REGISTRY_PORT:-5001}
+declare -r CONFIG_OUT_DIR=${CONFIG_OUT_DIR:-"_output/generated-manifest"}
```

Comment:
I don't think we generate any manifests while setting up cluster.

Reply:
should env file after those lines here?

Reply:
No, the env file is meant to set these values as shown below
```
#.env 

CTR_CMD=podman
CLUSTER_PROVIDER=microshift
CONFIG_OUT_DIR=/tmp/kepler/output
```

```
./main.sh up
```

It will deploy microshift and use podman

How does this work?

1. main.sh will read the `.env` file as the first thing, so these variables are set 
2. the script then runs 
```
declare -r CTR_CMD=${CTR_CMD:-docker}
declare -r CLUSTER_PROVIDER=${CLUSTER_PROVIDER:-kind}
declare -r REGISTRY_PORT=${REGISTRY_PORT:-5001}
declare -r CONFIG_OUT_DIR=${CONFIG_OUT_DIR:-"_output/generated-manifest"}
```

The above does 2 things 
1. makes the variables readonly from now on 
2. see if there the value is already set? and if it is set, use that value, else (`:-`) use the default provided.

I hope this clarifies the usage